### PR TITLE
Pbatko/benchtool stream runtime criterium

### PR DESCRIPTION
**Pull request by pbatko-da**

### CODE REVIEW

File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/AllocatedParties.scala [hunk 1]
```diff
@@ -6,16 +6,20 @@ package com.daml.ledger.api.benchtool.submission
...
 ) {
   val allAllocatedParties: List[Primitive.Party] =
-    List(signatory) ++ observers ++ divulgees ++ extraSubmitters ++ observerPartySetO.fold(
+    signatoryO.toList ++ observers ++ divulgees ++ extraSubmitters ++ observerPartySetO.fold(
       List.empty[Primitive.Party]
     )(_.parties)
+
+  /** NOTE: This is safe to call only you're sure signatory exists
+    */
+  def signatory: Primitive.Party = signatoryO.get
```
Comment (pbatko-da) [inline ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/submission/AllocatedParties.scala:22]: NOTE: This is being addressed on a preceding PR now: https://github.com/digital-asset/daml/pull/14244

File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/MetricsSet.scala [hunk 2]
```diff
@@ -27,7 +30,7 @@ object MetricsSet {
       recordTimeFunction = _.transactions.collect {
         case t if t.effectiveAt.isDefined => t.getEffectiveAt
       },
-      objectives = objectives,
+      configO = objectives,
```
Comment (tudor-da) [inline ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/MetricsSet.scala:33]: it was closer to the truth with `objectives` no?

**Review by tudor-da — APPROVED** [review]

File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/metrics/TotalStreamRuntimeMetric.scala [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  */
+case class TotalStreamRuntimeMetric[T](
+    clock: Clock,
+    startTime: Instant,
+    objective: MaxDurationObjective,
+) extends Metric[T] {
+  override type V = Value
+  override type Objective = MaxDurationObjective
+
+  private var lastSeenItemTime: Instant = startTime
```
[the remark this answers is not in the document]
Comment (pbatko-da) [reply ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/metrics/TotalStreamRuntimeMetric.scala:45]: This metric, and thus this var, is used solely as internal state of an actor defined in` MetricsCollector.handlingMessages` 
so I think there's no need for adding a volatile marker.

https://doc.akka.io/docs/akka/2.6.18/general/jmm.html
> In layman’s terms this means that changes to internal fields of the actor are visible when the next message is processed by that actor. So fields in your actor need not be volatile or equivalent.

File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/MetricsSet.scala (hunk 2 above)
Comment (pbatko-da) [reply ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/MetricsSet.scala:33]: It was confusing because it was as if we're creating objectives from objectives while more truthful is that we're creating objectives from their config.
I could name it `objectivesConfigO` but that's already given by the config type `configO: Option[TransactionObjectives]`

File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/metrics/TotalStreamRuntimeMetric.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (tudor-da) [reply ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/metrics/TotalStreamRuntimeMetric.scala:47]: true if you consider the outer scope of this class. Fine to keep it as is 👍 Thanks for the note

File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/MetricsSet.scala (hunk 2 above)
Comment (tudor-da) [reply ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/MetricsSet.scala:33]: no biggie, but it just looked strange to see it changed from `objectives = objectives` to `configO = objectives`. I would have expected both to be adapted or none

Comment (pbatko-da) [reply ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/metrics/MetricsSet.scala]: Changing in both places
